File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.py
def solution(a, b):  
    return getPoint(a, b)


def getPoint(a, b):
    point = 0
    if a%2 == 1 and b%2 ==1:
        point = a**2+b**2
    elif a%2 == 1 or b%2 ==1:
        point = 2*(a+b)
    else:
        point = abs(a-b)
    return point

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) { 
        switch(getNumberType(a, b)) {
            case 0:
                return Math.abs(a-b);
            case 1:
                return 2*(a+b);
            case 2:
                return (int)(Math.pow(a, 2) + Math.pow(b, 2));
        }   
        return 0;
    }
    
    public int getNumberType(int a, int b) {
        int cnt = 0;
        if(a%2 != 0) cnt ++;
        if(b%2 != 0) cnt ++;
        return cnt;
    }
}